torch/_dynamo/utils.py

```diff
@@ -620,19 +660,32 @@ class CompilationMetrics:
     has_guarded_code: bool
 
 
+@dataclasses.dataclass
+class BwdCompilationMetrics:
+    compile_id: str
+    inductor_compile_time_s: Optional[float]
+    code_gen_time_s: Optional[float]
+
+
 DEFAULT_COMPILATION_METRICS_LIMIT = 64
 
 
-_compilation_metrics: Deque[CompilationMetrics] = collections.deque(
-    maxlen=DEFAULT_COMPILATION_METRICS_LIMIT
-)
+_compilation_metrics: Deque[
+    Union[FwdCompilationMetrics, BwdCompilationMetrics]
+] = collections.deque(maxlen=DEFAULT_COMPILATION_METRICS_LIMIT)
 
 
-def record_compilation_metrics(compilation_metrics: CompilationMetrics):
+def record_compilation_metrics(
+    compilation_metrics: Union[FwdCompilationMetrics, BwdCompilationMetrics]
+):
     global _compilation_metrics
     _compilation_metrics.append(compilation_metrics)
+    if isinstance(compilation_metrics, FwdCompilationMetrics):
+        name = "fwd_compilation_metrics"
+    else:
+        name = "bwd_compilation_metrics"
     torch._logging.trace_structured(
-        "compilation_metrics",
+        name,
```

Comment:
I'm meh on this. In particular, tlparse currently knows how to parse `compilation_metrics` but not `fwd_compilation_metrics`. If you rename it this way, you have to adjust tlparse to know how to deal with both namings. But why not just not rename it.

Reply:
Ditto for the backwards naming. Like, you don't *have* to keep the names the same (tlparse is smart enough to ignore sections that it doesn't understand), but if you do rename it, you need to update tlparse

Reply:
The main reason of having two separate metrics definition is there are too many Nones at backward metrics if we use the original ```CompilationMetrics```. But right now, I was convinced to not rename it and keep both fwd and bwd use the same definition class.

Reply:
Well hang on. What does the name here affect whether or not you have two separate dataclasses. I also think two separate dataclasses is appropriate lol